addons/website_sale/static/src/website_builder/products_item_option.js

```diff
@@ -18,20 +18,36 @@ export class ProductsItemOption extends BaseOptionComponent {
             itemSize: this.props.itemSize,
         });
 
+        this.getProductsGridTable = () => {
+            return this.env.getEditingElement().closest(".o_wsale_products_grid_table");
+        };
+
+        this.domState = useDomState(() => {
+
+            // If /shop page layout is list, do not display Size option
+            const displaySizeOption = !this.getProductsGridTable()?.classList.contains("o_wsale_products_opt_layout_list");
+
+            if(displaySizeOption && this.state.itemSize) {
+                this.addClassToTableCells(this.state.itemSize.x, this.state.itemSize.y, "selected");
+            }
+
+            return {
+                displaySizeOption,
+            }
+        });
+
         onWillStart(async () => {
             this.defaultSort = await this.props.loadInfo();
 
             // need to display "re-order" option only if shop_default_sort is 'website_sequence asc'
             this.displayReOrder = this.defaultSort[0].shop_default_sort === "website_sequence asc";
-
-            // If /shop page layout is list, do not display Size option
-            const wsale_container = this.env.getEditingElement().closest("#o_wsale_container");
-            this.displaySizeOption = !wsale_container.classList.contains("o_wsale_products_opt_layout_list");
-            this.maxWidth = parseInt(wsale_container.dataset.ppr) || 5;
+            const computedStyle = window.getComputedStyle(this.getProductsGridTable());
```

Comment:
`window` => mmm we can't really avoid using it, right 🤔 ?

I think that all these custom properties are set inline. If it's the case, this should be enought:
```js
const pprValue = this.productsGridTableEl.style.getPropertyValue('--o-wsale-ppr').trim();
```

If it's not doable, can we at least do something like the following?
Please note that in my examples I've used a variable rather than a function (see my prev. comments)

```js
const targetWindow = this.productsGridTableEl.ownerDocument.defaultView || window;
const computedStyle = targetWindow.getComputedStyle(this.productsGridTableEl);
```